### PR TITLE
feat: reflect Renku 2.0 search in the URL

### LINKED ISSUE #3157: Search in Renku 2.0 should be reflected in the browser's URL
Currently, the search state is stored in a redux slice, which is not the appropriate location for search filters. Search URLs should be copy-pastable and the search state should be reflected in the browser's URL.

### PR DESCRIPTION
**Pull request by leafty**
Closes #3157.

The search page in Renku 2.0 has been reworked to make the search query and search options be reflected in the browser's URL.

/deploy

### CODE REVIEW

**Review by andre-code — COMMENTED** [review]
Excellent work, @leafty! 👏 The code functions as intended and is very clear and clean. I have only a few minor suggestions.

File: client/src/features/searchV2/components/SearchV2DateFilters.tsx [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+          type="radio"
+        />
+        <label className="form-check-label" htmlFor={id}>
+          Custom
+        </label>
+      </div>
+      {isChecked && (
+        <>
+          <div>
+            <label className={cx("px-2", "author-label")}>From:</label>
```
[the remark this answers is not in the document]
Comment (leafty) [reply client/src/features/searchV2/components/SearchV2DateFilters.tsx]: Removed

**Review by andre-code — DISMISSED** [review]
🚀 LGTM

**Comment by lorenzo-cavazzi** [comment]
I'm finishing a review on this, hold on a few minutes :pray: 
Sorry for the delay

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
Great work! Just a few comments before merging (see inline).

* Nitpick: when clicking on the filters, the line `<n> results for <filters>` immediately shows the new selection for the filters, but the number is updated later, when the API response arrives (I hope it's visible on the low-quality video). That's not a big deal, but it would be nice to either update them at the same time (it's fine if it's a little delayed) or show a loader instead of the number while waiting for the API response
Peek 2024-08-06 11-13
* The application seems to refresh entirely when changing the search parameters (see video where re-drawn components flash). I assume there aren't other ways since we are changing the URL but the WebSocket channel gets re-created every time. I'm wondering if there is any way to address this.
Peek 2024-08-06 13-44

File: client/src/features/searchV2/components/SearchV2DateFilters.tsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+          type="radio"
+        />
+        <label className="form-check-label" htmlFor={id}>
+          Custom
+        </label>
+      </div>
+      {isChecked && (
+        <>
+          <div>
+            <label className="px-2">From:</label>
```
[the remark this answers is not in the document]
Comment (leafty) [reply client/src/features/searchV2/components/SearchV2DateFilters.tsx:259]: This is keeping code you originally wrote for date filters -> https://github.com/SwissDataScienceCenter/renku-ui/commit/302737fe6297303aa831d06a2dec31416133126e#diff-2fd2bbc01a81ab8190494a0f096c095ce1c17bf250f42c692f4d6bb099dd149f. 🤷‍♀️

[the remark this answers is not in the document]
Comment (lorenzo-cavazzi) [reply client/src/features/searchV2/components/SearchV2DateFilters.tsx:259]: Ouch! Do you mind removing the class in both places? I didn't do that when updating the design :sweat:

**Comment by leafty** [comment]
> * Nitpick: when clicking on the filters, the line `<n> results for <filters>` immediately shows the new selection for the filters, but the number is updated later, when the API response arrives (I hope it's visible on the low-quality video). That's not a big deal, but it would be nice to either update them at the same time (it's fine if it's a little delayed) or show a loader instead of the number while waiting for the API response

Done

> * The application seems to refresh entirely when changing the search parameters (see video where re-drawn components flash). I assume there aren't other ways since we are changing the URL but the WebSocket channel gets re-created every time. I'm wondering if there is any way to address this.

Nothing we can do until we refactor how the app is bootstrapped (`src/index`, `src/App`).

**Review by lorenzo-cavazzi — APPROVED** [review]
Lgtm! :rocket:

**Pull request merged**